src/sql/src/plan/statement.rs

```diff
@@ -626,6 +636,22 @@ fn handle_show_create_sink(
     }
 }
 
+fn handle_show_create_index(
+    scx: &StatementContext,
+    name: ObjectName,
+) -> Result<Plan, failure::Error> {
+    let name = scx.resolve_item(name)?;
+    let entry = scx.catalog.get_item(&name);
+    if let CatalogItemType::Index = entry.item_type() {
```

Comment:
Can just use `if` rather than `if let`, right? Or does/can this enum not implement `Eq`?

Reply:
Yeah--this is something that's done widely in the code base that I don't really get. This was a line cargo-culted from another implementation that I didn't bother to change. Would love to clean these up in the future, but probably not worth another trip through CI today.